frontend/packages/sdk/src/contracts/PinkContract.ts

```diff
@@ -326,75 +345,78 @@ export class PinkContractPromise<
     options: PinkContractOptions,
     params: unknown[]
   ): SubmittableExtrinsic<'promise'> => {
-    const api = this.api as ApiPromise
+    const command = options.plain ? PlainInkCommand : EncryptedInkCommand
+    const payload = command(
+      this.contractKey,
+      this.abi.findMessage(messageOrId).toU8a(params),
+      options.value,
+      convertWeight(options.gasLimit || BN_ZERO).v2Weight,
+      options.storageDepositLimit
+    )
+    return this.api.tx.phalaPhatContracts
+      .pushContractMessage(this.address, payload.toHex(), options.deposit || BN_ZERO)
+      .withResultTransform((result: ISubmittableResult) => {
+        return new PinkContractSubmittableResult(
+          this.phatRegistry,
+          result,
+          applyOnEvent(result, ['ContractEmitted', 'ContractExecution'], (records: EventRecord[]) => {
+            return records
+              .map(
+                ({
+                  event: {
+                    data: [, data],
+                  },
+                }): DecodedEvent | null => {
+                  try {
+                    return this.abi.decodeEvent(data as Bytes)
+                  } catch (error) {
+                    console.error(`Unable to decode contract event: ${(error as Error).message}`)
+                    return null
+                  }
+                }
+              )
+              .filter((decoded): decoded is DecodedEvent => !!decoded)
+          })
+        )
+      })
+  }
 
-    // Generate a keypair for encryption
-    // NOTE: each instance only has a pre-generated pair now, it maybe better to generate a new keypair every time encrypting
-    const seed = hexToU8a(hexAddPrefix(randomHex(32)))
-    const pair = sr25519KeypairFromSeed(seed)
-    const [sk, pk] = [pair.slice(0, 64), pair.slice(64)]
+  public async send(messageOrId: string, options: PinkContractSendOptions, ...args: unknown[]) {
```

Comment:
The `#inkCommand` function is synchronous, so we cannot include the `autoDeposit` function inside it because the estimating process is asynchronous. We added a new function call `send` here, it include the `autoDepost`:
```ts
const result = await contract.send('newBadge', { pair, autoDeposit: true }, name)
await result.waitFinalized()
```